### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Daniil-Pankieiev**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,46 @@
...
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        else:
+            return 0
```
[the remark this answers is not in the document]
Comment (Daniil-Pankieiev) [reply app/main.py:20]: Else has no sense here. if self.is_hungry is True, the first return statement will be executed, and if it's False, the second return statement will be executed

File: app/main.py [hunk 2]
```diff
@@ -1 +1,46 @@
...
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(hungry_animals: list) -> int:
+    resutlt = 0
+    for animal in hungry_animals:
+        if animal.is_hungry:
+            resutlt += animal.feed()
+    return resutlt
```
[the remark this answers is not in the document]
Comment (Daniil-Pankieiev) [reply app/main.py:46]: Like this?

**Review by kostomeister — APPROVED** [review]
good jobs

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1 +1,45 @@
...
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, 7, is_hungry)
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(hungry_animals: list) -> int:
```
[the remark this answers is not in the document]
Comment (Daniil-Pankieiev) [reply app/main.py]: Sounds logical) Fixed it

File: app/main.py [hunk 4]
```diff
@@ -1 +1,45 @@
...
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list) -> int:
+    result = sum(
+        animal.feed() for animal in animals
+    )
```
Comment (LiudmylaKulinchenko) [inline app/main.py:44]: You can skip splitting into lines if it fits in one

**Review by LiudmylaKulinchenko — APPROVED** [review]
Nice! 🎈🍬😋

File: app/main.py (hunk 4 above)
Comment (Daniil-Pankieiev) [reply app/main.py]: Fixed

**Pull request closed without merging**